dev/NumberBox/NumberBox.cpp

```diff
@@ -0,0 +1,555 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "common.h"
+#include "NumberBox.h"
+#include "RuntimeProfiler.h"
+#include "ResourceAccessor.h"
+
+
+NumberBox::NumberBox()
+{
+    __RP_Marker_ClassById(RuntimeProfiler::ProfId_NumberBox);
+    SetDefaultStyleKey(this);
+}
+
+void NumberBox::OnApplyTemplate()
+{
+    // Initializations - Visual Components
+    winrt::IControlProtected controlProtected = *this;
+    m_TextBox = GetTemplateChildT<winrt::TextBox>(L"InputBox", controlProtected);
+    m_SpinDown = GetTemplateChildT<winrt::RepeatButton>(L"DownSpinButton", controlProtected);
+    m_SpinUp = GetTemplateChildT<winrt::RepeatButton>(L"UpSpinButton", controlProtected);
+    m_WarningIcon = GetTemplateChildT<winrt::FontIcon>(L"ValidationIcon", controlProtected);
+    m_ErrorTextMessage = GetTemplateChildT<winrt::TextBlock>(L"ErrorTextMessage", controlProtected);
+
+    // Initializations - Visual States
+    SetSpinButtonVisualState();
+    SetHeader();
+    SetPlaceHolderText();
+    
+    // Initializations - Interactions
+    m_ErrorToolTip.Content(m_ErrorToolTipTextBlock);
+    winrt::ToolTipService::SetToolTip(m_WarningIcon, m_ErrorToolTip);
+
+    if (m_SpinDown)
+    {
+        m_SpinDown.Click({ this, &NumberBox::OnSpinDownClick });
```

Comment:
Attaching to events needs to detach the handlers later to avoid dangling back pointers. Use the winrt::auto_revoke pattern found elsewhere in the codebase.